llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -236,10 +263,70 @@ getRootDescriptorsBindingInfo(const mcdxbc::RootSignatureDesc &RSD,
   return RDs;
 }
 
-static void validateRootSignature(Module &M,
-                                  const mcdxbc::RootSignatureDesc &RSD,
-                                  dxil::ModuleMetadataInfo &MMI,
-                                  DXILResourceMap &DRM) {
+static void validateDescriptorTables(Module &M,
+                                     const mcdxbc::RootSignatureDesc &RSD) {
```

Comment:
```suggestion
static void validateDescriptorTable(Module &M,
                                     const mcdxbc::DescriptorTable &Table) {
```

Does it make more sense to call `validateDescriptorTable` when we encounter a descriptor table while iterating in `validateRootSignature`?